### PR TITLE
Typescript and flatten everything b/c it's not 2014

### PR DESCRIPTION
**Pull request by jquense**

### CODE REVIEW

**Comment by jquense** [comment]
also includes #45

**Comment by jquense** [comment]
cc @TrySound

File: .babelrc [hunk 1]
```diff
@@ -1,3 +1,3 @@
 {
-  "presets": ["@react-bootstrap/babel-preset"]
+  "presets": ["@react-bootstrap", "@babel/typescript"]
```
Comment (TrySound) [inline .babelrc]: How can it be `@react-bootstrap`?

File: src/closest.ts [hunk 2]
```diff
@@ -0,0 +1,15 @@
+import matches from './matches'
+
+let isDoc = (obj: any) => obj != null && obj.nodeType === 9
+
+export default function closest(
+  node: Element,
+  selector: string,
+  context: Element
+): Element | undefined {
+  while (node && (isDoc(node) || !matches(node, selector))) {
```
Comment (TrySound) [inline src/closest.ts]: Do you know there is native well supported `element.closest`?

File: src/scrollParent.ts [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+
+export default function scrollPrarent(node: HTMLElement) {
+  let position = css(node, 'position')
+  let excludeStatic = position === 'absolute'
+  let ownerDoc = node.ownerDocument
+
+  if (position === 'fixed') return ownerDoc || document
+
+  // @ts-ignore
+  while ((node = node.parentNode) && node.nodeType !== 9) {
```
Comment (taion) [inline src/scrollParent.ts]: use constant?

File: src/closest.ts [hunk 4]
```diff
@@ -0,0 +1,15 @@
+import matches from './matches'
+
+let isDoc = (obj: any) => obj != null && obj.nodeType === 9
```
Comment (taion) [inline src/closest.ts]: use constant?

File: src/isWindow.ts [hunk 5]
```diff
@@ -0,0 +1,11 @@
+export default function isWindow(
+  node: Element | Document | Window
+): Window | false {
+  if ('window' in node && node.window === node) return node
+  if ('nodeType' in node && node.nodeType === 9) {
```
Comment (taion) [inline src/isWindow.ts]: use constant?

**Comment by jquense** [comment]
It probably doesn't make sense to support as many old things as we have here but I think that's worth doing in a follow up. Most of the bulky bits are ie8 cruft

File: .babelrc (hunk 1 above)
Comment (jquense) [reply .babelrc]: It's a shorthand Babel understands, eslint work similarly as well. Essentially just appends 'babel-preset' to the end when it sees a lone scope

**Pull request merged**